### PR TITLE
refactor(blooms): Extract planner queue into pkg

### PR DESCRIPTION
**Pull request by salvacorts**
**What this PR does / why we need it**:

We are going to wrap the queue to store tasks in the filesystem. We start by extracting the queue into a new pkg. Note that we the current logic is the same as before. We'll implement the disk storage in a followup PR.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: pkg/bloombuild/planner/queue/queue.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+type Task interface {
+	Tenant() string
+	Table() string
+	ID() string
+}
+
+// Queue is a wrapper of queue.RequestQueue that uses the file system to store the pending tasks.
+// When a task is enqueued, it's stored in the file system and recorded ad pending.
+// When it's dequeued, it's removed from the queue but kept in FS until removed.
+type Queue[T Task] struct {
```
Comment (chaudum) [inline pkg/bloombuild/planner/queue/queue.go:27]: We should not make the mistake of making the package generic. It solves a very specific use-case, and is also located in the `pkg/bloombuild/planner/` package.

Generics make code more difficult to manage than needed

Comment (salvacorts) [reply pkg/bloombuild/planner/queue/queue.go]: Got too fancy there. Removed the generics.

File: pkg/bloombuild/planner/queue/queue.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+func (q *Queue) RegisterConsumerConnection(consumer string) {
+	q.queue.RegisterConsumerConnection(consumer)
+}
+func (q *Queue) UnregisterConsumerConnection(consumer string) {
+	q.queue.UnregisterConsumerConnection(consumer)
+}
+
+// Enqueue adds a task to the queue.
+func (q *Queue) Enqueue(tenant string, task Task, successFn func()) error {
```
Comment (chaudum) [inline pkg/bloombuild/planner/queue/queue.go:105]: Do we still need the interface when the Queue is not generic? Can we use `*QueueTask` instead, and move the QueueTask into this package?

```suggestion
func (q *Queue) Enqueue(tenant string, task *QueueTask, successFn func()) error {
```

Comment (salvacorts) [reply pkg/bloombuild/planner/queue/queue.go:105]: Not a big deal but:
- We set the `queueTime` only once when we add it to the queue for the first time.
- The queueTask has a `resultsChannel` field that has nothing to do with the queue, but with the rest of the queue impl.

I think  we should leave the queue where it's now. But no strong opinions.

Comment (chaudum) [reply pkg/bloombuild/planner/queue/queue.go:105]: Ok with it.

**Review by chaudum — APPROVED** [review]

**Pull request merged**